src/main/scala/io/smartdatalake/util/historization/Historization.scala

```diff
@@ -145,21 +137,25 @@ private[smartdatalake] object Historization extends SmartDataLakeLogger {
       case (None, None) => newFeedDf.columns.sorted.toSeq
       case (Some(_), Some(_)) => throw new ConfigurationException("historizeWhitelist and historizeBlacklist mustn't be used at the same time.")
     }
-    val newFeedHashed = newFeedDf.withColumn("hash", hashFunc(struct(historizeHashCols.map(col): _*)))
+
+    // Generic column expression to generate a JSON string of the hash fields
+    def colHashExpr(cols: Seq[String]): Column = hash(to_json(struct(cols.map(col): _*)))
```

Comment:
What is the benefit of using to_json hier? The names of the columns can't change inside a DataFrame. In my opinion this just eats performance... but maybe you have a good reason?
I suggest to used directly "hash(cols.map(col): _*)"

Reply:
Hi @zzeekk ,

See the example in the issue. The structure of the dataframe doesn't change, but which columns are null can do and therefore the solution using values only can miss version changes. Agreed we increase the payload for the hash function but I think that's a price worth paying for accuracy.

Reply:
...maybe the hash is redundant?